### DISCUSSION TITLE
how to use plugin global properties ?

### DISCUSSION THREAD

**Discussion by kenan-dfrn** [nuxt/nuxt#17819] [Questions]
Hello,

i want to create a service called Resources in order to store all my api request set this service as globalProperties $resources in my plugin (to use it everywhere with this.$resources). btw i got this error : Cannot set property '$resources' of undefined

do you know if this is a bug or if there is another way ?

services/resources
```
import axios from "axios"

class Resource {
  constructor () {
    this.http = axios // to do custom htpp plugin
  }

  async getRscLanguages () {
    const {data} = await this.http.get(`http://localhost:8083/api/rsc/languages`)
    return data
  }
}

export default Resource
```

plugin/resources
```
import Resources from '../services/resources'

export default defineNuxtPlugin(nuxtApp => {
  nuxtApp.globalProperties.$resources = 'test'
  // nuxtApp.vueApp.config.globalProperties.$resources = Resources
})
```

**Comment by TheAlexLichter** [comment]
You can try using `provide`. Check out [the docs](https://v3.nuxtjs.org/docs/usage/nuxt-app#providing-helpers) on this topic as well.

```ts
// plugin
export default defineNuxtPlugin(nuxtApp => {
  nuxtApp.provide('hello', (name) => `Hello ${name}!`)
})

// in your app
nuxtApp.$hello('name')
